.github/workflows/current-workflow.yml

```diff
@@ -50,6 +50,47 @@ jobs:
         run: docker rm flask_backend_container
         
       - run: echo "Docker-and-Python-Flask-Service job's status is ${{ job.status }}."
+
+  Publish-Flask-Image-to-Docker-Hub:
+    name: Publish Docker Python Flask Image to Docker Hub
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+      attestations: write
+      id-token: write
+    steps:
+      - run: echo "The job was automatically triggered by a ${{ github.event_name }} event."
+      - run: echo "Job is running on a ${{ runner.os }} server hosted by GitHub"
+      - run: echo "Branch name is ${{ github.ref }} and repository is ${{ github.repository }}."
+      - name: Check out repository code
+        uses: actions/checkout@v4
+      - run: echo "The repository ${{ github.repository }} has been cloned to runner."
+      - run: echo "Workflow is now ready to test code on runner."
+
+      - name: Install required dependencies
+        run: |
+          cd ./code/backend
+          pip install -r requirements.txt
+
+      - name: Build the Docker Image
+        working-directory: ./code/backend
+        run: docker build -t flask_backend .
+      
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+        
+      - name: Tag flask_backend container image
+        run: docker tag flask_backend rzheng2024/seprojects-cs673olf24team6:${{ github.sha }}_back_end
```

Comment:
You are using your username and own personal GitHub sha to handle this.
This sha and username is unique only to you and only you have access to it.
This is also really bad for you and your machine from a security perspective so we have to have another way to handle this that will work on all machines and users.

**To set this up so it works on all machines:**
you need to use the username and password you created on lines 83 and 84
then change line 87 to something like docker push {${secrets.Docker_USERNAME}}/sep........

**What other need to do to ensure this works**
Then people will need to set their own Docker Hub credentials.
**So this means we will all need our own docker account** as well so that we can login and create a Docker Hub user name and access token to be used in GitHub so that people will be able to push an image once the pipeline has started.
Once that token is created we need to add it to github in a specific format.